### ISSUE TITLE
Queries on BYOL implementation

### ISSUE THREAD

**Issue by ankitpatnala** [PyTorchLightning/lightning-bolts#708]
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
In the papers it talks about LARS optimizer, but here the implementation was on Adam
Final thing, while optimizing with LARS, they have mentioned to not apply LARS adaptation and weight decay to batch norm and biases. It is not applied here.
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
![image](https://user-images.githubusercontent.com/26856470/130419483-8ee5ad99-6f00-46f2-876d-055d13155500.png)
The loss function mentined in the paper is 2-2*F.cosine_similarity but in implementation, i found it  -2* F.cosine_similarity ? Is it being handled somehwere else in the code ?

Please let me know whether my observations are correct and yet the code needs to be improved ?
Right now, i am using this piece of code for implementing in remote sensing images and it is collapsing? 
What could be the reason ? ( 4 channels, Adam optimizer (3e-4*batch_size/256), rest all parameters i am taking same as default
